### PR TITLE
UI for no insights in export dialog

### PR DESCRIPTION
**Pull request by oleg-morozov**
JIRA: TNT-622

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7458/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9309/ : SUCCESS in 9m 59s

File: libs/sdk-ui-dashboard/src/presentation/localization/bundles/en-US.json [hunk 1]
```diff
@@ -255,6 +255,11 @@
         "comment": "Label of a button that opens attachment selection",
         "limit": 0
     },
+    "dialogs.schedule.email.attachment.select.disabled.message": {
+        "value": "This dashboard does not contain any insights.",
```
Comment (scavnickyj) [inline libs/sdk-ui-dashboard/src/presentation/localization/bundles/en-US.json:259]: We need to also add "report" variant of the translation.

File: libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/ScheduledMailDialogRenderer/Attachments/AttachmentsSelectionDropdown.tsx [hunk 2]
```diff
@@ -66,6 +66,8 @@ const AttachmentsSelectionDropdownComponent: React.FC<IAttachmentsSelectionDropd
             onCancel={handleOnCancel}
             buttonClassName="s-schedule-select-attachments-button"
             bodyClassName="s-schedule-select-attachments-body"
+            bubbleMessageId="dialogs.schedule.email.attachment.select.disabled.message"
```
Comment (scavnickyj) [inline libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/ScheduledMailDialogRenderer/Attachments/AttachmentsSelectionDropdown.tsx:69]: This prop seems unnecessary

File: libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/ScheduledMailDialogRenderer/Attachments/ScheduleDropdown.tsx [hunk 3]
```diff
@@ -91,15 +100,31 @@ export const ScheduleDropdownComponent: React.FC<IScheduleDropdownProps> = (prop
...
+                        }}
+                    >
+                        {iconComponent && (
+                            <div className="gd-schedule-dropdown-button-icon">{iconComponent}</div>
+                        )}
+                        {title}
+                    </a>
+                    {buttonDisabled && bubbleMessageId && (
+                        <Bubble className="bubble-primary" alignPoints={ALIGN_POINTS}>
+                            <FormattedMessage id={bubbleMessageId} />
```
Comment (scavnickyj) [inline libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/ScheduledMailDialogRenderer/Attachments/ScheduleDropdown.tsx:124]: The translation id here does not need to be passed from parent, right? Let's make it inline if there is no logic and spare the component one prop.

File: libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/ScheduledMailDialogRenderer/Attachments/ScheduleDropdown.tsx [hunk 4]
```diff
@@ -91,15 +100,31 @@ export const ScheduleDropdownComponent: React.FC<IScheduleDropdownProps> = (prop
...
+                        className={cx(
+                            "gd-schedule-dropdown-button",
+                            buttonClassName,
+                            buttonDisabled && "gd-schedule-dropdown-button-disabled",
+                        )}
+                        target="_blank"
+                        rel="noopener noreferrer"
+                        onClick={() => {
+                            if (buttonDisabled) return;
+                            toggleDropdown();
```
Comment (scavnickyj) [inline libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailDialog/ScheduledMailDialogRenderer/Attachments/ScheduleDropdown.tsx:114]: maybe we can do `!buttonDisabled && toggleDropdown()`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7474/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9327/ : SUCCESS in 10m 16s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7475/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9328/ : SUCCESS in 9m 44s

**Review by scavnickyj — APPROVED** [review]

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2404

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2163/ : SUCCESS in 1m 10s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2449/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1272/ : SUCCESS in 18m 42s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9334/ : SUCCESS in 9m 55s

**Pull request merged**